src/vm/variables.rs

```diff
@@ -50,11 +50,18 @@ pub fn lookup_reserved_variable(
                 Ok(Some(sender))
             }
             NativeVariables::ContractCaller => {
-                let sender = env
+                let caller = env
                     .caller
                     .clone()
-                    .ok_or(RuntimeErrorType::NoSenderInContext)?;
-                Ok(Some(sender))
+                    .ok_or(RuntimeErrorType::NoCallerInContext)?;
+                Ok(Some(caller))
+            }
+            NativeVariables::TxSponsor => {
+                let sponsor = env
+                    .sponsor
+                    .clone()
+                    .ok_or(RuntimeErrorType::NoSponsorInContext)?;
```

Comment:
I have a double on my understanding here - are we throwing a runtime error when a sponsor can not be found?

Reply:
Good point. I'll change this.